controllers/storagecluster/provider_server.go

```diff
@@ -62,10 +62,16 @@ func (o *ocsProviderServer) ensureDeleted(r *StorageClusterReconciler, instance
 		return reconcile.Result{}, err
 	}
 
+	serviceType := corev1.ServiceTypeClusterIP
+	if instance.Spec.ProviderAPIServerServiceType != nil {
+		serviceType = *instance.Spec.ProviderAPIServerServiceType
+	} else if instance.Spec.HostNetwork {
+		serviceType = corev1.ServiceTypeNodePort
+	}
 	var finalErr error
 
 	for _, resource := range []client.Object{
-		GetProviderAPIServerService(instance),
+		GetProviderAPIServerService(instance.Namespace, serviceType),
 		GetProviderAPIServerDeployment(instance),
```

Comment:
We don't need to explicitly delete deployment and services as storageCluster owns both. Should we remove this logic?

Reply:
can be done, I'm keeping the changes to the minimum until we test the whole flow atleast once and then start cleaning up.